File: entregavel/fibonacci.java
import java.util.Scanner;

public class Fibonacci {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Quantos números da sequência de Fibonacci você quer ver? ");
        int termos = scanner.nextInt();

        int primeiro = 0;
        int segundo = 1;

        System.out.println("Sequência de Fibonacci com " + termos + " termos:");

        for (int i = 1; i <= termos; i++) {
            System.out.print(primeiro + " ");

            // A mágica acontece aqui: calculamos o próximo e atualizamos as variáveis
            int proximo = primeiro + segundo;
            primeiro = segundo;
            segundo = proximo;
        }

        System.out.println(); // Apenas quebra a linha no final
        scanner.close();
    }
}

File: entregavel/fibonacci.js
const readline = require('readline');

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

rl.question("Quantos números da sequência de Fibonacci você quer ver? ", (input) => {
    const termos = parseInt(input);

    let primeiro = 0;
    let segundo = 1;
    
    // Criamos uma lista (array) para guardar a sequência e printar bonito no final
    const sequencia = [];

    for (let i = 1; i <= termos; i++) {
        sequencia.push(primeiro); // Adiciona o número atual na lista

        let proximo = primeiro + segundo;
        primeiro = segundo;
        segundo = proximo;
    }

    // .join(", ") junta todos os números do array separados por uma vírgula e espaço
    console.log(`Sequência de Fibonacci: ${sequencia.join(", ")}`);

    rl.close();
});
